lib/OpenQA/WebAPI/Controller/Admin/Workers.pm

```diff
@@ -90,7 +90,7 @@ sub previous_jobs_ajax ($self) {
                         id => $job_id,
                         name => $job->name,
                         deps => $job->dependencies,
-                        result => $job->result,
+                        result => ($job->result // 'none') eq 'none' ? undef : $job->result,
```

Comment:
You could've use constants here.

Reply:
Ah right.
I think passing `result` in every case (also in the "All tests" queries) could be even a better solution...